syntax = "proto3";

service DominionServer {
    rpc Join(PlayerInfo) returns (stream Message) {}
    rpc PlayActionCard(Card) returns(Response) {}
    rpc Buy(Card) returns(Response) {}
    rpc EndTurn(PlayerInfo) returns(Response) {}
}

message PlayerInfo {
    string name = 1;
}

message Card {
    string name = 1;
}

message Message {
    string type = 1;
    string data = 2;
}

message Response {
    bool ok = 1;
}